sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/output/BatchReportReader.java

```diff
@@ -130,6 +131,15 @@ public Issues readDeletedComponentIssues(int deletedComponentRef) {
     return null;
   }
 
+  @CheckForNull
+  public FileStream readSourceLines(int fileRef) {
+    File file = fileStructure.fileFor(FileStructure.Domain.SOURCE, fileRef);
+    if (doesFileExists(file)) {
+      return new FileStream(file);
+    }
+    return null;
```

Comment:
Should we return null if no source find ? Will all component of type FILE have source ? If yes, I think we should fail here, WDYT ?